File: warmup-02/python/problem_E.py
if __name__ == "__main__":

	for _ in range(int(input())):
		n = int(input())
		strr = input()

		encode = "abcdefghijklmnop"
		position = 0

		for i in range(0,n,4):
			if strr[i]=='1':
				position+=8
			if strr[i+1]=='1':
                		position+=4
            		if strr[i+2]=='1':
                		position+=2
            		if strr[i+3]=='1':
                		position+=1
        		print(encode[position],end='')
        		position=0
        	print()

File: warmup-02/cpp/problem_E.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int t; scanf("%d",&t);

    while(t--){
        int n;
        string str;
        cin >> n >> str; // input n & str


        string encode="abcdefghijklmnop";
        int position=0;

        for(int i=0;i<n;i+=4){
            if(str[i]=='1')
                position+=8;
            if(str[i+1]=='1')
                position+=4;
            if(str[i+2]=='1')
                position+=2;
            if(str[i+3]=='1')
                position+=1;

            cout<<encode[position];
            position=0;
        }
        printf("\n");
    }
    return 0;
}
